### DISCUSSION TITLE
Runing a file on client only?

### DISCUSSION THREAD

**Discussion by MariuzM** [solidjs/solid-start#1039] [Q&A]
I have api file that is checking local storage for token, can this hole file be run on client only because i get error `localStorage is not defined`

**Comment by jcalfee** [comment]
Same here.  I have a directive checking for light dark color themes ..  I'm trying to figure out how to run that client-side only without putting the entire app to `ssr: false`.  I did see a feature in another framework to set SSR on a per-route basis.  That is still a hack but that would be pretty useful and a lot better then disabling ssr for the entire app.

**Comment by jcalfee** [comment]
I was able to prevent errors on `localStorage` and `window` (in my case) by surrounding by the isServer check (`import { isServer } from 'solid-js/web'`).  That should work in your case.  There is still a need to shield the 1st render from the server as well, in my case, I get a light colored page that flips to dark mode after it is seen in the media or localStorage that the theme should be dark.
